### ISSUE TITLE
Eval bug: gemma4uv mmproj + CUDA → SIGABRT in mtmd_helper_decode_image_chunk → llama_context::decode (workarounds from #24251 and #24314 do not help)

### ISSUE THREAD

**Issue by hsjlyj** [ggml-org/llama.cpp#26981]
### Name and Version

```
$ llama-server --version
version: b10375 (533fa7f9)
built with ... for Linux x86_64 (CUDA build, RTX 4090)
```

### Operating systems

Linux (Ubuntu 22.04.5 LTS container, kernel 6.12)

### GGML backends

CUDA

### Hardware

- CPU: AMD EPYC 7502 x128
- RAM: 1 TiB system
- GPU: 1 x NVIDIA RTX 4090 (24 GiB VRAM, driver 580.x, CUDA 12.x)
- 32 GiB free VRAM at the moment of the crash

### Models

- Text: `Gemma4-31B-QAT-Uncensored-HauhauCS-Balanced-Q4_K_M.gguf` (18.7 GB)
- Vision: `mmproj-Gemma4-31B-QAT-Uncensored-HauhauCS-Balanced-BF16.gguf` (1.2 GB)
- Draft MTP: `mtp-gemma-4-31B-it.gguf` (280 MB) — **NOT loaded in this repro; the bug reproduces with MTP disabled.**

### Problem description

Loading any image via the OpenAI-compatible `/v1/chat/completions` endpoint
crashes the server with SIGABRT, even after applying the workarounds
recommended in issues #24251 and #24314.

Reproduction:

```bash
llama-server \
  --model /root/rivermind-data/llm-models/hauhaucs-gemma4-31b-qat-uncensored-mtp/Gemma4-31B-QAT-Uncensored-HauhauCS-Balanced-Q4_K_M.gguf \
  --mmproj /root/rivermind-data/llm-models/hauhaucs-gemma4-31b-qat-uncensored-mtp/mmproj-Gemma4-31B-QAT-Uncensored-HauhauCS-Balanced-BF16.gguf \
  --no-mmproj-offload \
  --no-warmup \
  --host 0.0.0.0 --port 8081 --api-key *** \
  --ctx-size 8192 --parallel 1 --cont-batching --batch-size 512 --ubatch-size 128 \
  -ngl 99 -fa on \
  --temp 0.6 --top-k 64 --top-p 0.9 --min-p 0.05 --repeat-penalty 1.1 \
  --no-webui --jinja --log-timestamps
```

(The same crash reproduces with `--spec-type draft-mtp --model-draft <mtp.gguf>`
added; MTP draft is orthogonal to this crash.)

Send a request with even a trivial 1x1 PNG image:

```bash
curl -H "Authorization: Bearer ***" -H "Content-Type: application/json" \
  -d '{"model":"hauhau-gemma4-31b-uncensored-mtp","messages":[{"role":"user","content":[
    {"type":"text","text":"What color? one word."},
    {"type":"image_url","image_url":{"url":"data:image/png;base64,iVBORw0KGgo..."}}
  ]}],"max_tokens":30}' \
  http://127.0.0.1:8081/v1/chat/completions
```

The server processes the first request OK and returns a response — but on a
*subsequent* image request, the process aborts. (If the first request itself
succeeds and only later requests fail, that's the failure mode I observed
twice in this session. The crash may also occur on the very first request if
the upstream is load-balancing; I have not narrowed this down further.)

### Stack trace

```
ggml_print_backtrace () from /opt/llama.cpp/b10375/build/bin/libggml-base.so.0
ggml_abort () from /opt/llama.cpp/b10375/build/bin/libggml-base.so.0
llama_context::decode(llama_batch const&) () from .../libllama.so.0
llama_decode () from .../libllama.so.0
mtmd_helper_decode_image_chunk () from .../libmtmd.so.0
server_slot::process_mtmd_chunk(unsigned long, unsigned long&)
server_context_impl::pre_decode()::{lambda(server_slot&)#5}::operator()(server_slot&)
server_context_impl::iterate(...)
server_context_impl::pre_decode()
server_context_impl::update_slots()
server_queue::start_loop(long)
```

### What I tried (and that did NOT help)

| Workaround | Source | Result |
| --- | --- | --- |
| `--no-mmproj-offload` (run vision encoder on CPU) | issue #24251 | Still crashes; same stack |
| Drop MTP draft (`--spec-type draft-mtp` removed) | issue #24314 | Still crashes; same stack |
| Combined both workarounds | this report | Still crashes; same stack |
| Smaller batch/ubatch (`-b 128 -ub 32`) | — | Still crashes; same stack |
| Disable flash attention (`-fa off`) | — | Crashes; same stack |
| `--parallel 1` (default already) | — | Crashes; same stack |
| 1x1 PNG input (simplest possible image) | — | Crashes; same stack |

The crash is NOT memory-related: GPU was 20 GiB / 24 GiB used, well within
limits. The crash is NOT MTP-related: reproduces with no draft model loaded.

### Workaround that DOES work

Drop `--mmproj` entirely. Vision is unavailable, but the text-only server is
stable. This is the only config I can run in production on this build.

### Expected Behavior

Loading Gemma 4 with its bundled `mmproj-...BF16.gguf` and sending an image
should produce a coherent response, with the vision encoder running on CPU
if that is the safer path for this architecture.

### Actual Behavior

SIGABRT in `mtmd_helper_decode_image_chunk → llama_context::decode`. Process
exits, GPU memory released, server stops listening on the port. Any pending
or subsequent requests (text or image) get connection refused.

### Relation to other issues

- #24251 (gemma4uv on GPU → NaN logits → sampler assert): different stack
  (sampler, not decode). Possibly a separate manifestation of the same root
  cause.
- #24314 (MTP + mmproj + flash-attn → memory aperture violation on
  Ada/Blackwell): different stack (fattn.cu, not llama_decode). Triggered by
  MTP draft; mine reproduces with MTP disabled.
- #25522 (Gemma 4 crashes with MTP): depends on MTP draft.
- #24324 (llama.cpp-b9568 fattn.cu:579): also depends on MTP draft.

So this is a fourth Gemma 4 vision crash path. They may share an upstream
cause — the gemma4uv vision encoder interacting poorly with CUDA graphs on
Ada/Blackwell — but I have not bisected.

### First Bad Commit

Not bisected. I only have b10375 to test on this machine. Issues #24251,
#24314, #25522 are all on builds ≤ b10375; issue #21402 cites b8639, b8650
already showing a similar load-time abort. The bug has been present for a
while.

### Hardware

```
$ nvidia-smi --query-gpu=name,memory.total,driver_version,compute_cap --format=csv
name, memory.total [MiB], driver_version, compute_cap
NVIDIA GeForce RTX 4090, 24564 MiB, 580.65.06, 8.9
```

Compute capability 8.9 (sm_89, Ada Lovelace). The patterns reported in
#24314 (memory aperture violation) and #24251 (NaN logits on GPU) suggest
this generation is particularly affected.
